### PR TITLE
Cherry-pick to 6.x: Add DeDot in add_docker_metadata processor (#9505)

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
From the original PR:
This pr is for issue: https://github.com/elastic/beats/issues/9350
Currently metrics from docker container is after dedot, which look like this:
> t  docker.container.labels.com_docker_compose_service	       	grafana
t  docker.container.labels.com_docker_compose_version	       	1.22.0

and metrics from postgresql is not dedotted, which look like:
> t  docker.container.labels.com.docker.compose.service	       	sentry-postgres
t  docker.container.labels.com.docker.compose.version	       	1.22.0

This PR will add a config option in add_docker_metadata processor to enable/disable dedot.

(cherry picked from commit 58573a9f3f154d41bb68596958b72e71c9ee41c8)

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
Travis CI failure is caused by https://github.com/elastic/beats/issues/9597

File: auditbeat/docs/getting-started.asciidoc [hunk 1]
```diff
@@ -97,6 +97,22 @@ endif::[]
 [[docker]]
 *docker:*
 
+ifeval::["{release-state}"=="unreleased"]
```
Comment (ruflin) [inline auditbeat/docs/getting-started.asciidoc]: How is this related to this PR?

File: metricbeat/module/elasticsearch/node/_meta/data.json [hunk 2]
```diff
@@ -47,4 +47,4 @@
     "service": {
         "name": "elasticsearch"
     }
-}
\ No newline at end of file
+}
```
Comment (ruflin) [inline metricbeat/module/elasticsearch/node/_meta/data.json]: Nit: This change probably happened because you opened this file manually but it does not belong into this PR.

Comment (kaiyan-sheng) [reply metricbeat/module/elasticsearch/node/_meta/data.json]: Yeah hmm and I thought it's just an extra line but after deleting the extra line, it still shows up different.
